### PR TITLE
docs: add new API getting started doc [DET-3348 DET-3368]

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

view generated (potentially stale) doc: https://133789-253846879-gh.circle-artifacts.com/0/docs/site/html/how-to/rest-apis.html#

- [ ] add link to the resulting doc from circleci artifacts.

## Commentary (optional)

This branch is on our main repo (and not on my fork) so everyone has write access. Feel free to push an update if you'd like to fix something : )

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/how-to/rest-apis.txt [hunk 1]
```diff
@@ -0,0 +1,95 @@
+.. _rest-api-getting-started:
+
+Getting Started with REST APIs
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:3]: This section describes how our APIs are made, and I think we instead want to jump right in to how to use them.  Maybe put the Swagger UI section first?  I would really focus on consumption by people who probably don't know what all of these tools we used to make the endpoints are.

File: docs/how-to/rest-apis.txt [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+it easy for back end implementation and client side consumption.
+
+
+
+Reach the Swagger UI for your Determined installation by click API icon from the
+Determined WebUI or by visiting the path `/swagger-ui` on your Determined cluster.
+
+.. image:: /assets/images/swagger-ui-location.png
+
+
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:37]: Add a sentence describing what you see when you find the Swagger UI

File: docs/how-to/rest-apis.txt [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+endpoint, currently at `/api/v1/auth/login` with the following format:
+
+.. code:: json
+
+  {
+    "username": "string",
+    "password": "string"
+  }
+
+Example:
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:55]: Could we provide more of a story example?  Something like:

For example, to retrieve details about an experiment run in Determined:

"example that hits the /expeirments endpoint"

**Review by davidhershey — CHANGES_REQUESTED** [review]
Good work overall and its written super clearly!  I think we should restructure this slightly to tell the story of how to interact with the APIs -- why you need them, what they can do, and how to use them.

**Comment by hamidzr** [comment]
I took a crack at addressing your comments. lmk what you think

File: docs/how-to/rest-apis.txt [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+Swagger UI allows anyone — be it your development team or your end consumers —
+to visualize and interact with the API’s resources without having any of the
+implementation logic in place. It’s automatically generated from your OpenAPI
+(formerly known as Swagger) Specification, with the visual documentation making
+it easy for back end implementation and client side consumption.
+
+
+
+Reach the Swagger UI for your Determined installation by click API icon from the
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:23]: 'by clicking the API icon'

File: docs/how-to/rest-apis.txt [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+Determined WebUI or by visiting the path `/swagger-ui` on your Determined cluster.
+
+.. image:: /assets/images/swagger-ui-location.png
+
+The Swagger UI gives you an up-to-date view of what APIs your Determined cluster is
+providing in a readable and interactive fashion.
+In this page you'll see a list of all the available endpoints grouped by their workflows.
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:30]: On this page

File: docs/how-to/rest-apis.txt [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        "agent_uid": 0,
+        "agent_gid": 0
+      }
+    }
+  }
+
+Now store and attach the provided token to your future API calls under
+`Authorization` header and with the following format: `Bearer $TOKEN`.
+
+
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:81]: I think I would create a new subheader here called "Example"

**Review by davidhershey — CHANGES_REQUESTED** [review]
Really like the new format, it looks great.  Just a few tiny changes this time.

File: docs/how-to/rest-apis.txt [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+We provide detailed documentation on all the available functionalities (aka endpoints) and
+alongside a playground to interact with them through Swagger UI.
+
+Swagger UI
+----------
+
+As Swagger puts it:
+
+Swagger UI allows anyone — be it your development team or your end consumers —
+to visualize and interact with the API’s resources without having any of the
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:16]: Is there some kind of quote box we can put this in to visually distinguish it from our text?

File: docs/how-to/rest-apis.txt [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  }
+
+Once we have the token we can store and attach to our future API calls under
+`Authorization` header and with the following format: `Bearer $TOKEN`.
+
+
+Example:
+--------
+
+For example, let's find an archived example to un-archive:
```
Comment (davidhershey) [inline docs/how-to/rest-apis.txt:87]: archived _experiment_

**Review by davidhershey — APPROVED** [review]
Looks really good -- just a few quick things but then should be good to go from my end

File: docs/how-to/rest-apis.txt (hunk 7 above)
Comment (hamidzr) [reply docs/how-to/rest-apis.txt]: it looks indenting it is the way to go https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks

**Comment by hamidzr** [comment]
tests are passing. I'll skip running our gpu tests as this is just documentation changes. https://determined-ai.slack.com/archives/CSG3W08JY/p1599860283027200

**Pull request merged**